controlplane/kubeadm/internal/controllers/remediation.go

```diff
@@ -191,9 +211,107 @@ func (r *KubeadmControlPlaneReconciler) reconcileUnhealthyMachines(ctx context.C
 
 	log.Info("Remediating unhealthy machine")
 	conditions.MarkFalse(machineToBeRemediated, clusterv1.MachineOwnerRemediatedCondition, clusterv1.RemediationInProgressReason, clusterv1.ConditionSeverityWarning, "")
+
+	// Set annotations tracking remediation is in progress (remediation will complete when a replacement machine is created).
+	annotations.AddAnnotations(controlPlane.KCP, map[string]string{
+		controlplanev1.RemediationInProgressAnnotation: "",
+	})
+
+	// Stores info about last remediation.
```

Comment:
I wonder if we should use one "reconciliationTime" instead of doing various time.Now calls which all have slightly different results